### PR TITLE
apply gitlabci-lint --fix

### PR DESCRIPTION
**Pull request by peczenyj**
most of the changes does not impact the code

I use this list of linters, then apply the --fix

I would not recomment update the `.golangci.yml` yet, this is something that needs a mode deeply discussion

```yaml
linters:
  enable:
    - asasalint
    - asciicheck
    - bidichk
    - bodyclose
    - containedctx
    - contextcheck
    - cyclop
    - deadcode
    - decorder
    - depguard
    - dogsled
    - dupl
    - durationcheck
    - errcheck
    - errchkjson
    - errname
    - errorlint
    - execinquery
    - exhaustive
    - exhaustruct
    - exportloopref
    - forbidigo
    - forcetypeassert
    - funlen
    - gochecknoglobals
    - gochecknoinits
    - gocognit
    - goconst
    - gocritic
    - gocyclo
    - godot
    - godox
    - goerr113
    - gofmt
    - gofumpt
    - goheader
    - goimports
    - gomnd
    - gomoddirectives
    - gomodguard
    - goprintffuncname
    - gosec
    - gosimple
    - govet
    - grouper
    - ifshort
    - importas
    - ineffassign
    - interfacebloat
    - ireturn
    - lll
    - logrlint
    - maintidx
    - makezero
    - misspell
    - nakedret
    - nestif
    - nilerr
    - nilnil
    - nlreturn
    - noctx
    - nolintlint
    - nosnakecase
    - nosprintfhostport
    - paralleltest
    - prealloc
    - predeclared
    - promlinter
    - reassign
    - rowserrcheck
    - sqlclosecheck
    - structcheck
    - tagliatelle
    - tenv
    - testpackage
    - thelper
    - tparallel
    - typecheck
    - unconvert
    - unparam
    - unused
    - usestdlibvars
    - varcheck
    - varnamelen
    - wastedassign
    - whitespace
    - wrapcheck
    - wsl

```

### CODE REVIEW

**Review by mihaitodor — COMMENTED** [review]
Thanks for looking into this! When I started the initial effort to bring in some linters, I tried to avoid the ones which seemed overly-restrictive, so it won't feel too nitpicky at first. Also, do these extra linters increase the golangci-lint run time much?

File: internal/docs/yaml_test.go [hunk 1]
```diff
@@ -281,20 +281,20 @@ e:
...
 				docs.FieldObject("foo", "").WithChildren(
-					docs.FieldObject("eles", "").Array().WithChildren(
+					docs.FieldObject("eels", "").Array().WithChildren(
 						docs.FieldString("bar", "").HasDefault("default"),
 					),
 				),
 			},
 			yaml: `
 foo:
-  eles:
```
[the remark this answers is not in the document]
Comment (peczenyj) [reply internal/docs/yaml_test.go:291]: oh it seems the linter misspell that thinkgs that the term `eles` was about `eels`

we can revert this specific file if you want, or substitute by something that have no meaning like 'lol'

[the remark this answers is not in the document]
Comment (mihaitodor) [reply internal/docs/yaml_test.go:291]: It's more of a generic concern that the linter which caused this change might be annoying in other similar situations in the future. If it's just one place, then yeah, maybe change it to `bar` or smth like that.

**Comment by peczenyj** [comment]
> Thanks for looking into this! When I started the initial effort to bring in some linters, I tried to avoid the ones which seemed overly-restrictive, so it won't feel too nitpicky at first. Also, do these extra linters increase the golangci-lint run time much?

@mihaitodor I did not update the list of linters, I am just telling what was the list used to apply the automatic fixes via option `--fix`

I do not recomment enable the entire list because it will be insane to review and fix (and most of them does not add a real value, only keeps the code more readable like lll that avoids long lines).

this can be done step by step if needed

**Comment by mihaitodor** [comment]
OK, but then the changes from this PR won't be enforced in the future via GitHub Actions, or I'm not sure I follow.

**Comment by peczenyj** [comment]
> OK, but then the changes from this PR won't be enforced in the future via GitHub Actions, or I'm not sure I follow.

No they will not enforced

File: public/service/processor.go [hunk 2]
```diff
@@ -185,7 +185,7 @@ func (o *OwnedProcessor) ProcessBatch(ctx context.Context, batch MessageBatch) (
 	return batches, nil
 }
 
-// Close the processor, allowing it to clean up resources. It is
+// Close the processor, allowing it to clean up resources. It is.
```
Comment (Jeffail) [inline public/service/processor.go:188]:
```suggestion
// Close the processor, allowing it to clean up resources.
```

**Review by Jeffail — CHANGES_REQUESTED** [review]
LGTM, some of the comments look a bit odd ending with a period but I don't mind swallowing the whole pill for the sake of the spelling corrections, which there are surprisingly many 😅

Just one suggestion, obviously an unfinished comment so might as well trim the whole second bit.

**Comment by peczenyj** [comment]
done

**Review by Jeffail — APPROVED** [review]

**Comment by Jeffail** [comment]
Thanks @peczenyj! I officially declare this repo typo and spelling error free... FOREVER.

**Pull request merged**